staging/src/k8s.io/apiserver/pkg/registry/generic/rest/streamer.go

```diff
@@ -81,3 +86,17 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	stream = resp.Body
 	return
 }
+
+func checkRedirect(req *http.Request, via []*http.Request) error {
+	// Copied from http defaultCheckRedirect.
+	if len(via) >= 10 {
+		return errors.New("stopped after 10 redirects")
+	}
+	// Only allow redirects to the same host.
+	newHost := req.URL.Host
+	originalHost := via[0].URL.Host
```

Comment:
is len(via) always greater than 0?

Reply:
Yes, a redirect always has an original request. See https://github.com/golang/go/blob/214f7ec554888042a7a54fdca9cba19aee8ebaf1/src/net/http/client.go#L551-L581